stellargraph/core/graph_networkx.py

```diff
@@ -745,21 +745,49 @@ def edges(self, triple=False) -> Iterable[Any]:
     def has_node(self, node: Any) -> bool:
         return self._graph.__contains__(node)
 
-    def neighbors(self, node: Any) -> Iterable[Any]:
+    def _transform_edges(self, edges, get_node, weight, edge_types):
+        edge_types_set = set(edge_types) if edge_types is not None else None
+
+        def get(e):
+            return (
+                (get_node(e), e[2].get(self._edge_weight_label))
+                if weight
+                else get_node(e)
+            )
+
+        def is_correct_type(e):
+            return (
+                edge_types_set is None
+                or e[2].get(self._edge_type_attr) in edge_types_set
+            )
+
+        return {get(e) for e in edges if is_correct_type(e)}
```

Comment:
Hmm ah I copied the existing behaviour without thinking about it, but yeah that makes sense. I guess we should return a list instead?

Reply:
I guess so, but maybe it should just be an issue that we address later, given we're likely to want to change the existing behaviour too?

Reply:
Okay sure, I've created #653

Reply:
@kjun9 & @huonw , not sure if the following behavior is related to the discussion above but seemed like it is, so adding the comment here: 
If an `edge_type` has multiple attributes, only one of the attributes accessible. Although the `StellarGraph` graph does list all the attributes in the `info()` method. In fact, its only the attributed with the name `weight` that one can access using `neighbours()`, `in_nodes()`, 'out_nodes()`, and `edge_weights()` methods.
Here is a short example: 
```
G = nx.MultiDiGraph()

G.add_edge('a', 'b',  wt = 3000, weight=6, label='temporal')
G.add_edge('a', 'c', weight=2, w = .001, label='temporal')
G.add_edge('c', 'd', weight=1, label='temporal')
G.add_edge('c', 'e', weight=7, label='temporal')
G.add_edge('c', 'f', weight=9, label='temporal')
G.add_edge('a', 'd', weight=3, label='temporal')

G.add_edge('b', 'f', closeness = 10, label='friend')
G.add_edge('c', 'd',  label='friend')

SG = sg.StellarGraph(G)
print(SG.info())

list(SG.neighbors('a',include_edge_weight=True))
list(SG.neighbors('b',include_edge_weight=True, edge_types = ['temporal']))

list(SG.neighbors('b',include_edge_weight=True, edge_types = ['friend']))
```

Currently, can't think of where multiple attributes of an `edge_type` would be useful but thought to point out the behavior just so that we know. 

I think the more critical point is whether the attribute has to have a name `weight` or it can be user specified and when `include_edge_weight` is set to `True` or `get_weights` is used, it returns the value of the attribute.

Reply:
The name of the weight label can currently be specified by the user using the `edge_weight_label` argument to `StellarGraph`, however this doesn't seem to be documented anywhere and so, until #659, wasn't visible in the docs at all.